src/util/filter.js

```diff
@@ -99,6 +99,7 @@ export function matchesFilter(filter: IgnoreFilter, basename: string, loc: strin
   }
   return filter.regex.test(loc) ||
          filter.regex.test(`/${loc}`) ||
+         filter.regex.test(`\\${loc}`) ||
```

Comment:
The argument to `test` is a string rather than a RegExp pattern, so it shouldn't be interpreting backslashes as the escape character in this context. Note that the following expression (on Windows) should produce a string whose first character is a backslash:

```js
`${path.sep}${loc}`
```

That said, I'm not sure why https://github.com/yarnpkg/yarn/pull/3175/commits/c2df043343092dcc408f8792ad16eb86cad6ba3a seemed to make things worse, test-wise (an additional test [failed](https://ci.appveyor.com/project/kittens/yarn/build/2195/job/q5u26f85qlroy533#L3011))